src/test/java/org/greenbuttonalliance/gbaresourceserver/usage/repository/IntervalBlockRepositoryTest.java

```diff
@@ -0,0 +1,200 @@
+				.published(LocalDateTime.parse("2012-03-02 05:00:00", SQL_FORMATTER))
+				.selfLinkHref(PRESENT_SELF_LINK)
+				.selfLinkRel("self")
+				.upLinkHref("/espi/1_1/resource/RetailCustomer/9B6C7066/UsagePoint/5446AF3F/MeterReading/01/IntervalBlock")
+				.upLinkRel("up")
+				.updated(LocalDateTime.parse("2012-03-02 05:00:00", SQL_FORMATTER))
+				.start(1330578000L)
+				.duration(1800)
+				.intervalReadings(Stream.of(
+					new IntervalReading()
+						.setCost(974L)
+						.setStart(1330578000L)
+						.setDuration(900)
+						.setValue(285L)
+						.setReadingQualities(
+							Stream.of(new ReadingQuality().setQuality(QualityOfReading.VALID),
+									new ReadingQuality().setQuality(QualityOfReading.RAW),
+									new ReadingQuality().setQuality(QualityOfReading.DERIVED))
+								.collect(Collectors.toSet())),
+					new IntervalReading()
+						.setCost(965L)
+						.setStart(1330578900L)
+						.setDuration(900)
+						.setValue(383L)
+						.setReadingQualities(
+							Stream.of(new ReadingQuality().setQuality(QualityOfReading.OTHER),
+									new ReadingQuality().setQuality(QualityOfReading.VALID))
+								.collect(Collectors.toSet())))
+					.collect(Collectors.toSet()))
+				.build(),
+			IntervalBlock.builder()
+				.published(LocalDateTime.parse("2012-03-03 05:00:00", SQL_FORMATTER))
+				.selfLinkHref("/espi/1_1/resource/RetailCustomer/9B6C7066/UsagePoint/5446AF3F/MeterReading/01/IntervalBlock/174")
+				.selfLinkRel("self")
+				.upLinkHref("/espi/1_1/resource/RetailCustomer/9B6C7066/UsagePoint/5446AF3F/MeterReading/01/IntervalBlock")
+				.upLinkRel("up")
+				.updated(LocalDateTime.parse("2012-03-03 05:00:00", SQL_FORMATTER))
+				.start(1330578800L)
+				.duration(900)
+				.intervalReadings(Stream.of(
+					new IntervalReading()
+						.setCost(922L)
+						.setStart(1330578800L)
+						.setDuration(900)
+						.setValue(350L)
+						.setReadingQualities(
+							Stream.of(new ReadingQuality().setQuality(QualityOfReading.VALID))
+								.collect(Collectors.toSet())))
+					.collect(Collectors.toSet()))
+				.build(),
+			IntervalBlock.builder()
+				.published(LocalDateTime.parse("2012-03-04 05:00:00", SQL_FORMATTER))
+				.selfLinkHref("/espi/1_1/resource/RetailCustomer/9B6C7066/UsagePoint/5446AF3F/MeterReading/01/IntervalBlock/175")
+				.selfLinkRel("self")
+				.upLinkHref("/espi/1_1/resource/RetailCustomer/9B6C7066/UsagePoint/5446AF3F/MeterReading/01/IntervalBlock")
+				.upLinkRel("up")
+				.updated(LocalDateTime.parse("2012-03-04 05:00:00", SQL_FORMATTER))
+				.start(1330987644L)
+				.duration(900)
+				.intervalReadings(Collections.emptySet())
+				.build()
+		);
+
+		// hydrate UUIDs and entity mappings
+		intervalBlocks.forEach(ib -> {
+			ib.setUuid(UuidCreator.getNameBasedSha1(UuidCreator.NAMESPACE_URL, ib.getSelfLinkHref()));
+			ib.getIntervalReadings().forEach(ir -> {
+				ir.setBlock(ib);
+				ir.getReadingQualities().forEach(rq -> rq.setReading(ir));
+			});
+		});
+		return intervalBlocks;
```

Comment:
This is also pretty messy, but I think there's also value in persisting these methods through the repository: saving these test objects to the database through the repository tests that functionality of the repository in itself.

Reply:
I agree.